produce/lib/produce/commands_generator.rb

```diff
@@ -146,8 +146,11 @@ def run
         FastlaneCore::CommanderGenerator.new.generate(Produce::Options.available_options, command: c)
 
         c.action do |args, options|
-          allowed_keys = Produce::Options.available_options.collect(&:key)
-          Produce.config = FastlaneCore::Configuration.create(Produce::Options.available_options, options.__hash__.select { |key, value| allowed_keys.include? key })
+          extra_options = [FastlaneCore::ConfigItem.new(key: :merchant_name),FastlaneCore::ConfigItem.new(key: :merchant_identifier)]
```

Comment:
@ohayon I've just answered in more detail in your other review comment on `options.rb`. It was removed so that the options do not show for unrelated commands, yet will still show for the merchant command